packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts

```diff
@@ -390,8 +390,8 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
   });
 
   const placeholder = React.useMemo(() => {
-    if (inPlaceholder) {
-      return inPlaceholder;
+    if (params.forwardedProps.hasOwnProperty('placeholder')) {
+      return inPlaceholder ?? '';
```

Comment:
Have you tried testing Flavien's suggestion?
I think that with it we could remove the `?? ''` here. 🤔

Reply:
And remove `params.forwardedProps` from the deps

Reply:
Oops 🙈 
Done in [d9d4964](https://github.com/mui/mui-x/pull/13148/commits/d9d4964ac1dac44567f810157a442b762e04d02f) and [78007e0](https://github.com/mui/mui-x/pull/13148/commits/78007e08cc9d8676bacb104eeba56bc0308b05ff)